scheduler/core/eventsqueue/event_queue.py

```diff
@@ -35,42 +38,35 @@ def add_event(self, event: Event) -> None:
 
 
 class EventQueue:
-    def __init__(self, night_indices: FrozenSet[NightIndex], sites: FrozenSet[Site]):
-        self._events = {night_idx: {site: NightEventQueue(night_idx=night_idx, site=site) for site in sites}
-                        for night_idx in night_indices}
+    def __init__(self, start: Time, end: Time, sites: FrozenSet[Site]):
+        curr = start.to_datetime()
+        self._events = {}
+        night_idx = 0
+        while curr <= end.to_datetime():
+            self._events[curr] = {site: NightEventQueue(night_idx=NightIndex(night_idx), site=site) for site in sites}
+            night_idx += 1
+            curr += timedelta(days=1)
+
         self._blockage_stack = []
 
-    def add_event(self, night_idx: NightIndex, site: Site, event: Event) -> None:
+    def add_event(self, night_date: datetime, site: Site, event: Event) -> None:
         match event:
-            case Blockage():
-                self._blockage_stack.append(event)
             case Interruption():
-                site_events = self.get_night_events(night_idx, site)
+                site_events = self.get_night_events(night_date, site)
                 if site_events is not None:
                     site_events.add_event(event)
                 else:
-                    raise KeyError(f'Could not add event {event} for night index {night_idx} to site {site.name}.')
-
-    def add_events(self, night_idx: NightIndex, site: Site, events: Iterable[Event]) -> None:
-        for event in events:
-            self.add_event(night_idx, site, event)
-
-    def check_blockage(self, resume_event: ResumeNight) -> Blockage:
-        if self._blockage_stack and len(self._blockage_stack) == 1:
-            b = self._blockage_stack.pop()
-            b.ends(resume_event.start)
-            return b
-        raise RuntimeError('Missing blockage for ResumeNight')
+                    raise KeyError(f'Could not add event {event} for date {night_date} to site {site.name}.')
 
-    def get_night_events(self, night_idx: NightIndex, site: Site) -> Optional[NightEventQueue]:
+    def get_night_events(self, night_date: datetime, site: Site) -> Optional[NightEventQueue]:
```

Comment:
Like here, we are already working with `NightIndex` in the code: it would be really nice to still have this as `NightIndex` and not changed to `night_date: datetime`, which is more concerning to me because it's `datetime` (what is the time?) instead of `datetime.date`.